### PR TITLE
Adding xarray-einstats feedstock.

### PR DESCRIPTION
**Pull request by ericmjl**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by OriolAbril** [comment]
I am willing and happy to be a maintainer of the feedstock

File: recipes/xarray-einstats/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  sha256: 5c4e00d09172daf665cbd5e533a897896b84f37feb5cab0fd75829477fbcda7d
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - pip
```
Comment (OriolAbril) [inline recipes/xarray-einstats/meta.yaml:20]: as the library has pyproject.toml only and no setup.py nor setup.cfg, it needs pip>=21.3, it also looks like flit should be installed for everything to work as it is a build system requirement (which I think means it should go here in the host)

Comment (ericmjl) [reply recipes/xarray-einstats/meta.yaml:20]: Great catch, thank you, @OriolAbril!

File: recipes/xarray-einstats/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - pip >=21.3
+    - python
```
Comment (OriolAbril) [inline recipes/xarray-einstats/meta.yaml:21]:
```suggestion
    - python >=3.7
    - flit
```

File: recipes/xarray-einstats/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  run:
+    - numpy >=1.21
+    - python
```
Comment (OriolAbril) [inline recipes/xarray-einstats/meta.yaml:24]:
```suggestion
    - python >=3.7
```

File: recipes/xarray-einstats/meta.yaml (hunk 2 above)
Comment (OriolAbril) [reply recipes/xarray-einstats/meta.yaml:21]: the azure jobs are complaining about flit, it is a build system dependency in pyproject: https://github.com/arviz-devs/xarray-einstats/blob/main/pyproject.toml#L2. I think it needs to be here like it is in the poliastro recipe for example: https://github.com/conda-forge/poliastro-feedstock/blob/main/recipe/meta.yaml

python could be bigger than 3.6 as the linter suggests, but the run is 3.7 so it is probably easier and less prone to confusion to use 3.7 in both

Comment (ericmjl) [reply recipes/xarray-einstats/meta.yaml]: Thanks for the catch, @OriolAbril! Okie just made the changes.

**Comment by ericmjl** [comment]
@OriolAbril can I check, is the license file distributed as part of the `pip`-installable package? I believe that may need to be added into the distributed source code in order for us to be able to do a conda-forge package.

**Comment by OriolAbril** [comment]
> @OriolAbril can I check, is the license file distributed as part of the pip-installable package? I believe that may need to be added into the distributed source code in order for us to be able to do a conda-forge package.

Not sure, let me see if I can check this somehow.

File: recipes/xarray-einstats/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+test:
+  imports:
+    - xarray_einstats
+  commands:
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://pypi.org/project/xarray-einstats/
```
Comment (OriolAbril) [inline recipes/xarray-einstats/meta.yaml:38]:
```suggestion
  home: https://github.com/arviz-devs/xarray-einstats
```

I think the arviz feedstock takes the license from github directly using the link provided here: https://github.com/conda-forge/arviz-feedstock/blob/main/recipe/meta.yaml#L41

Comment (OriolAbril) [reply recipes/xarray-einstats/meta.yaml:38]: I can try to add the license to pip package too but I'd need an example of how to do it (I tried to copy arviz and saw arviz doesn't include it in the pip package)

Comment (ericmjl) [reply recipes/xarray-einstats/meta.yaml:38]: I think I found an example to copy, @OriolAbril! 

`pyjanitor` uses a MANIFEST.in file to declare which other files need to be packaged together and distributed. 

When I looked up other references, I stumbled upon the rules by which Python package files are included: https://packaging.python.org/en/latest/guides/using-manifest-in/. Perhaps that could help as well.

**Comment by OriolAbril** [comment]
Preparing a patch release with a manifest file in https://github.com/arviz-devs/xarray-einstats/pull/12

**Comment by OriolAbril** [comment]
Should be fixed now with 0.2.2 release, thanks for the pointers.

Can you add https://xarray-einstats.readthedocs.io as [`doc_url`](https://github.com/conda-forge/arviz-feedstock/blob/main/recipe/meta.yaml#L52)?

**Review by xhochy — APPROVED** [review]

**Pull request merged**